ts-tests/tests/test-eth-balance.ts

```diff
@@ -68,7 +84,65 @@ async function eth_link(api: ApiPromise, alice: KeyringPair) {
 
   console.log(`\nStep 1: Link Ethereum account`);
 
-  const transaction = api.tx.accountLinkerModule.link(alice.address, 0, 0, 0, test_r, test_s, test_v);
+  const registry = new TypeRegistry();
+
+  // Encode prefix with concatenated utf8, instead of SCALE codec to match the litentry node implementation
+	let encodedPrefix = Buffer.from(msgPrefix, 'utf-8');
+  
+  //let encodedAccId = registry.createType('AccountId', alice.address).toU8a();
+  //console.log(encodedAccId);
+  //console.log(alice.addressRaw);
+
+  let encodedExpiredBlock = new UInt(registry, 10000, 32).toU8a();
+
+  let encodedMsg = new Uint8Array(encodedPrefix.length + alice.addressRaw.length + encodedExpiredBlock.length);
+  encodedMsg.set(encodedPrefix);
+  encodedMsg.set(alice.addressRaw, encodedPrefix.length);
+  encodedMsg.set(encodedExpiredBlock, encodedPrefix.length + alice.addressRaw.length);
+
+  // To use manual hash and sign method, a prefix of \x19Ethereum ... is also needed to be prefixed manually
+	//let hash = Hash.keccak256s(encodedMsg);
+
+  //console.log('hash is:');
+  //console.log(hash);
+
+	// TODO ECDSA keyring from polkadot crypto still not working
+  //const ropstenTestAcc = keyringRopsten.addFromUri('0xe82c0c4259710bb0d6cf9f9e8d0ad73419c1278a14d375e5ca691e7618103011');
+  //console.log('ropsten pub key: ');
+  //console.log(ropstenTestAcc.publicKey);
+  //console.log(ropstenTestAcc.address);
+  //console.log(crypto.keccakAsU8a(ropstenTestAcc.publicKey));
+  //let signedMsg = ropstenTestAcc.sign(new Buffer(hash2.slice(2), "hex"));
+  //let r = signedMsg.slice(0, 32);
+  //let s = signedMsg.slice(32, 64);
+  //let v = signedMsg[64];
+	
+  //var signature = Account.sign(hash, privateKey);
+  //var vrs = Account.decodeSignature(signature);
+  //console.log("signature is :");
+  //console.log(signature);
+  //console.log(keyPair.sign(new Buffer(hash2.slice(2), "hex"), { canonical: true }).r.toString(16));
+
+	// TODO Web3 could be used to replace eth-lib once ethereum prefix is implemented on account-linker side
+	const Web3 = require("web3");
+  const web3 = new Web3();
+   // Convert byte array to hex string
+  let hexString = "0x" + Buffer.from(encodedMsg).toString('hex');
+
+  let signedMsg = web3.eth.accounts.sign(hexString, privateKey);
+
+	// This is not needed as eth-lib already does the same job
+	//let keyPair = ec.keyFromPrivate(new Buffer(privateKey.slice(2), "hex"));
+	//let privKey = keyPair.getPrivate("hex");
+	//let pubKey = keyPair.getPublic();
+	//console.log(`Private key: ${privKey}`);
+	//console.log("Public key :", pubKey.encode("hex").substr(2));
+	//console.log("Public key (compressed):",
+	//    pubKey.encodeCompressed("hex"));
+  // let signature = ec.sign(hash, privKey, "hex", {canonical: true});
+
+  //const transaction = api.tx.accountLinkerModule.link(alice.address, 0, 10000, vrs[1], vrs[2], vrs[0]);
+  const transaction = api.tx.accountLinkerModule.linkEth(alice.address, 0, 10000, signedMsg.r, signedMsg.s, signedMsg.v);
```

Comment:
So in ts link_eth becomes linkEth and account_balance becomes accountBalance?

Reply:
yes, it's like this. if you want to double check the extrinsic name, you can check it directly on polkadot js apps, and the name it displays there matches what I use here.